### PR TITLE
Delete DOMTimestamp and EpochTimestamp

### PR DESCRIPTION
**Pull request by teoli2003**
These are not real interfaces but a kind of WebIDL-only typedef. We wanted to remove them for a long time.

This PR adapts all occurrences of `DOMTimestamp` and `EpochTimestamp`.

This is part of the cleaning needed for having the whole of Web/API covered with `page-type:` YAML headers. (See #16255)

### CODE REVIEW

File: files/en-us/web/api/cookiechangeevent/changed/index.md [hunk 1]
```diff
@@ -27,7 +27,7 @@ An array of objects containing the changed cookie(s). Each object contains the f
 - `path`
   - : A string containing the path of the cookie.
 - `expires`
-  - : A {{domxref("DOMTimeStamp")}} containing the expiration date of the cookie.
+  - : A timestamp (in milliseconds, excluding leap seconds, relative to 01 January, 1970 00:00:00 UTC) containing the expiration date of the cookie.
```
Comment (wbamberg) [inline files/en-us/web/api/cookiechangeevent/changed/index.md:30]: This is Unix time, given in ms, isn't it? What do you think about having a glossary entry for Unix time, and instead of writing it out here, saying:

> A timestamp given as [Unix time]() in milliseconds ...

...or something like that?

Comment (teoli2003) [reply files/en-us/web/api/cookiechangeevent/changed/index.md:30]: I don't like the "Unix time" expression (it is the Unix epoch though).

What about having a glossary entry about "timestamp" and writing:

> A [timestamp](https://github.com/mdn/content/pull/17096) in milliseconds …

Then, later, for `DOMHighResTimestamp` we can have a second one ("high resolution timestamp"):

> A [high resolution timestamp](https://github.com/mdn/content/pull/17096)

Of course, each glossary entry will link to the other.

@wbamberg: What do you think? This will allow us to hide the details, keeping them one click away, and to the trained readers it will be still obvious which of the timestamp is in action at that moment.

Comment (wbamberg) [reply files/en-us/web/api/cookiechangeevent/changed/index.md:30]: Why don't you like "Unix time"? It is a [real thing](https://en.wikipedia.org/wiki/Unix_time) isn't it? 

I think my issue with "timestamp" is that just a timestamp doesn't say anything about the definition, while "Unix time" is defined as seconds (or ms in this case) since the start of 1970.

Comment (teoli2003) [reply files/en-us/web/api/cookiechangeevent/changed/index.md:30]: Yes, you are right, I will use Unix time and add a glossary entry. (For whatever reason I never used that term, only the Unix epoch)

**Comment by teoli2003** [comment]
@wbamberg I created the glossary entry (which will work for the high-resolution timestamp too) and updated the sentence.

(I had to do a force push, sorry...)

File: files/en-us/web/api/cookiechangeevent/changed/index.md (hunk 1 above)
Comment (teoli2003) [reply files/en-us/web/api/cookiechangeevent/changed/index.md]: Done.

File: files/en-us/glossary/unix_time/index.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  - Glossary
+---
+The **Unix time** is the amount of time, expressed in millisecond, that happened since a specific point in time, called the _epoch_. It doesn't take into account leap seconds.
+
+It is a timestamp used to store a specific time. The epoch is arbitrarily set as January 1st, 1970, at midnight (UTC).
+
+The Unix time can be stored in two ways, depending on the need of the API:
+
+- As an integer.
+- As a floating-point value. The fractional part of the number indicates fractions of a millisecond, with a precision of up to 5 microseconds. Browsers _fuzz_ the value (typically between 60 microseconds and 2 milliseconds. This fuzzing increases security by preventing timing attacks.
```
Comment (wbamberg) [inline files/en-us/glossary/unix_time/index.md:14]: It might be helpful here to name these two methods. In particular if we called the second one "high-resolution timestamp" is would be more descriptive and link more directly to the type. But then I'm not sure what we would call the other one. A normal timestamp?

I'm also not sure "fuzz" is correct here: usually in security I think of fuzzing as providing bad or randomized values to an API, to stress-test it (https://developer.mozilla.org/en-US/docs/Glossary/Fuzzing). In any case "fuzz" is more jargony than just "reduced precision" which (I think?) is what we ware talking about here.

I'm not sure I understand "typically between 60 microseconds and 2 milliseconds", but I guess it refers to https://developer.mozilla.org/en-US/docs/Web/API/DOMHighResTimeStamp#reduced_time_precision, but that seems to be quite Firefox-specific.

So perhaps:

```suggestion
- A normal timestamp is stored as an integer.
- A high-resolution timestamp is stored as a floating-point value. The fractional part of the number indicates fractions of a millisecond, with a precision of up to 5 microseconds. Browsers may provide a coarser time precision for these timestamps, as a security measure to prevent timing attacks.
```

That said it does look like the coarsening algorithm is in the spec: https://w3c.github.io/hr-time/#dfn-coarsen-time so I wonder if we should describe it here. And then I'm a bit worried that this doesn't belong in a glossary entry for Unix time any more, and maybe we should instead include it in a section in https://developer.mozilla.org/en-US/docs/Web/API/Performance .

**Review by wbamberg — DISMISSED** [review]
Thanks! I had a few comments on the glossary entry. As said there I wonder if we should include some of the high-res timestamp stuff (especially the reduced resolution bits) somewhere under https://developer.mozilla.org/en-US/docs/Web/API/Performance , as it seems rather specific to an API and not very well-suited to that entry.

**Comment by bsmth** [comment]
Hi @teoli2003 I was about to update this one from `main` but there's a few conflicts, do you want to take a look? 😄

**Comment by hamishwillee** [comment]
@teoli2003 This needs a rebase and remaining fixes. Would be great to get in.

**Comment by wbamberg** [comment]
@hamishwillee , I think @Elchi3 and I are going to have to take care of this anyway as part of the work on the Performance API: https://github.com/openwebdocs/project/issues/62.

**Comment by hamishwillee** [comment]
@wbamberg Re https://github.com/orgs/mdn/discussions/282#discussioncomment-4141217

> > other specs - it is generally a unix epoch timestamp, but it doesn't have to be.
> 
> This I don't understand though, I thought `DOMTimeStamp` was replaced with `EpochTimeStamp` (according to https://developer.mozilla.org/en-US/docs/Web/API/DOMTimeStamp, and also features like https://developer.mozilla.org/en-US/docs/Web/API/Notification/timestamp, which we document as using `DOMTimeStamp`, but the spec uses `EpochTimeStamp`: https://notifications.spec.whatwg.org/#timestamp).

See the note in [DOMTimeStamp](https://developer.mozilla.org/en-US/docs/Web/API/DOMTimeStamp):

image

So yes, it has been removed and renamed in one spec, but it still gets reference in others. Lazy Hamish says it probably doesn't matter all that much - ie. those other specs are not all that important. But I have not done the investigation.

**Comment by wbamberg** [comment]
I've fixed the merge conflicts here and made the changes I recommended in my own review :). So this PR now does three things:

- adds a new glossary entry "Unix time"
- deletes the pages for `DOMTimeStamp` and `EpochTimeStamp`, and made them redirect to the glossary entry
- changes all links to `DOMTimeStamp` and `EpochTimeStamp` to links to the glossary entry

This doesn't yet tackle `DOMHighResTimeStamp` - I think that does not belong in a "Unix time" glossary entry, but in a bit of guide material for the Perf APIs, and it will be cleaner to do it in a different PR. When we do that, we should update the glossary entry to point lost people at this new guide material, like "if you are looking for high-res (sub-millisecond) timestamps used in the perf APIs, see XYZ".

File: files/en-us/mozilla/firefox/releases/23/index.md [hunk 3]
```diff
@@ -36,7 +36,7 @@ tags:
 - D3E [`KeyboardEvent.key`](/en-US/docs/Web/API/KeyboardEvent#key_names_and_char_values) is now supported, but only for non-printable keys ({{bug(842927)}}).
 - The `title` attribute of {{domxref("DOMImplementation.createHTMLDocument")}} is now optional as per updated DOM specification.
 - The ability to add a sidebar panel (`window.sidebar.addPanel`) has been dropped ({{bug(691647)}}).
-- The unprefixed {{domxref("Window.requestAnimationFrame")}} and {{domxref("Window.cancelAnimationFrame")}} methods has been added ({{bug(704063)}}). The unprefixed {{domxref("Window.requestAnimationFrame")}} receives a {{domxref("DOMHighResTimeStamp")}} as argument; the prefixed version did, and still is, received a {{domxref("DOMTimeStamp")}}({{bug(753453)}}).
+- The unprefixed {{domxref("Window.requestAnimationFrame")}} and {{domxref("Window.cancelAnimationFrame")}} methods has been added ({{bug(704063)}}). The unprefixed {{domxref("Window.requestAnimationFrame")}} receives a {{domxref("DOMHighResTimeStamp")}} as argument; the prefixed version did, and still is, received a timestamp in milliseconds ({{bug(753453)}}).
```
Comment (hamishwillee) [inline files/en-us/mozilla/firefox/releases/23/index.md:39]:
```suggestion
- The unprefixed {{domxref("Window.requestAnimationFrame")}} and {{domxref("Window.cancelAnimationFrame")}} methods has been added ({{bug(704063)}}). The unprefixed {{domxref("Window.requestAnimationFrame")}} receives a {{domxref("DOMHighResTimeStamp")}} as argument; the prefixed version receives a timestamp in milliseconds ({{bug(753453)}}).
```

File: files/en-us/mozilla/firefox/releases/54/index.md [hunk 4]
```diff
@@ -44,7 +44,7 @@ Firefox 54 was released on June 13, 2017. This article lists key changes that ar
 - The {{domxref("URLSearchParams.URLSearchParams", "URLSearchParams()")}} constructor now accepts a record containing strings as an init object ({{bug("1331580")}}).
 - Values returned in {{domxref("KeyboardEvent.key")}} for printable keys when the control key is also pressed have been corrected on macOS (except when the Command key is pressed) ({{bug(1342865)}}).
 - The `dom.workers.latestJSVersion` preference, which was mainly implemented to work around problems using [`let`](/en-US/docs/Web/JavaScript/Reference/Statements/let) in workers (due to {{bug(855665)}}, which has since been fixed) has been removed (see {{bug(1219523)}}).
-- The {{domxref("event.timeStamp")}} property now returns a high-resolution monotonic time ({{domxref("DOMHighResTimeStamp")}}) instead of an epoch time ({{domxref("DOMTimeStamp")}}).
+- The {{domxref("event.timeStamp")}} property now returns a high-resolution monotonic time ({{domxref("DOMHighResTimeStamp")}}) instead of an epoch time in milliseconds.
```
Comment (hamishwillee) [inline files/en-us/mozilla/firefox/releases/54/index.md:47]: Should these be linked to the unix time glossary? I don't think it matters particularly because these are old versions, but epoch time could mean anything.

File: files/en-us/mozilla/firefox/releases/23/index.md (hunk 3 above)
Comment (wbamberg) [reply files/en-us/mozilla/firefox/releases/23/index.md]: I just couldn't be bothered to fix Firefox 23 release notes.

File: files/en-us/web/api/geolocationcoordinates/longitude/index.md [hunk 5]
```diff
@@ -19,13 +19,14 @@ browser-compat: api.GeolocationCoordinates.longitude
...
-{{domxref("GeolocationPosition")}} interface, which is the object type returned by
-Geolocation API functions that obtain and return a geographical position.
+The {{domxref("GeolocationCoordinates")}} interface's read-only **`longitude`** property is
+a double-precision floating point value which represents the longitude of a geographical position,
+specified in decimal degrees.
+Together with a timestamp, given as [Unix time](/en-US/docs/Glossary/Unix_time) in milliseconds,
+indicating a time of measurement,
+the `GeolocationCoordinates` object is part of the {{domxref("GeolocationPosition")}} interface,
+which is the object type returned by Geolocation API functions
+that obtain and return a geographical position.
```
Comment (hamishwillee) [inline files/en-us/web/api/geolocationcoordinates/longitude/index.md:29]:
```suggestion
The {{domxref("GeolocationCoordinates")}} interface's read-only **`longitude`** property is a double-precision floating point value which represents the longitude of a geographical position, specified in decimal degrees.
Together with a timestamp, given as [Unix time](/en-US/docs/Glossary/Unix_time) in milliseconds, indicating a time of measurement, the `GeolocationCoordinates` object is part of the {{domxref("GeolocationPosition")}} interface, which is the object type returned by Geolocation API functions that obtain and return a geographical position.
```

File: files/en-us/web/api/geolocationcoordinates/longitude/index.md [hunk 6]
```diff
@@ -19,13 +19,8 @@ browser-compat: api.GeolocationCoordinates.longitude
...
 {{securecontext_header}}{{APIRef("Geolocation API")}}
 
-The {{domxref("GeolocationCoordinates")}} interface's read-only
-**`longitude`** property is a double-precision floating point
-value which represents the longitude of a geographical position, specified in decimal
-degrees. Together with a {{domxref("DOMTimeStamp")}} indicating a time of measurement,
-the `GeolocationCoordinates` object is part of the
-{{domxref("GeolocationPosition")}} interface, which is the object type returned by
-Geolocation API functions that obtain and return a geographical position.
+The {{domxref("GeolocationCoordinates")}} interface's read-only **`longitude`** property is a double-precision floating point value which represents the longitude of a geographical position, specified in decimal degrees.
```
Comment (hamishwillee) [inline files/en-us/web/api/geolocationcoordinates/longitude/index.md:22]: Perhaps not relevant to this change, but does it matter that it is double-precision floating point? Isn't it just a number in JavaScript

File: files/en-us/web/api/serviceworkerregistration/shownotification/index.md [hunk 7]
```diff
@@ -97,9 +97,10 @@ showNotification(title, options)
       - : An ID for a given notification that allows you to find,
         replace, or remove the notification using a script if necessary.
     - `timestamp`
-      - : A {{domxref("DOMTimeStamp")}} representing the time when
-        the notification was created. It can be used to indicate the time at which a
-        notification is actual. For example, this could be in the past when a notification
+      - : A timestamp, given as [Unix time](/en-US/docs/Glossary/Unix_time) in milliseconds,
+        representing the time when the notification was created.
+        It can be used to indicate the time at which a notification is actual.
+        For example, this could be in the past when a notification
```
Comment (hamishwillee) [inline files/en-us/web/api/serviceworkerregistration/shownotification/index.md:103]:
```suggestion
        For example, this could be in the past when a notification.
```

**Review by hamishwillee — APPROVED** [review]
Excellent. Just a couple of comments. If you either respond or mark those resolved I am happy to merge (or for you to).

File: files/en-us/mozilla/firefox/releases/23/index.md (hunk 3 above)
Comment (hamishwillee) [reply files/en-us/mozilla/firefox/releases/23/index.md]: I know. There's several minutes you'll never get back. OK with me.

File: files/en-us/web/api/geolocationcoordinates/longitude/index.md (hunk 6 above)
Comment (wbamberg) [inline files/en-us/web/api/geolocationcoordinates/longitude/index.md:22]:
```suggestion
The {{domxref("GeolocationCoordinates")}} interface's read-only **`longitude`** property is a number which represents the longitude of a geographical position, specified in decimal degrees.
```

Yes, I think so (although this is out of scope).

File: files/en-us/web/api/serviceworkerregistration/shownotification/index.md (hunk 7 above)
Comment (wbamberg) [reply files/en-us/web/api/serviceworkerregistration/shownotification/index.md]: Oh hang on, this is wrong.

File: files/en-us/web/api/serviceworkerregistration/shownotification/index.md [hunk 8]
```diff
@@ -97,11 +97,7 @@ showNotification(title, options)
       - : An ID for a given notification that allows you to find,
         replace, or remove the notification using a script if necessary.
     - `timestamp`
-      - : A {{domxref("DOMTimeStamp")}} representing the time when
-        the notification was created. It can be used to indicate the time at which a
-        notification is actual. For example, this could be in the past when a notification
-        is used for a message that couldn't immediately be delivered because the device
-        was offline, or in the future for a meeting that is about to start.
+      - : A timestamp, given as [Unix time](/en-US/docs/Glossary/Unix_time) in milliseconds, representing the time associated with the notification. This could be in the past when a notification is used for a message that couldn't immediately be delivered because the device was offline, or in the future for a meeting that is about to start.
```
Comment (wbamberg) [inline files/en-us/web/api/serviceworkerregistration/shownotification/index.md:100]: I thought "the time at which a notification is actual." was a typo but it is a quote from the spec: https://notifications.spec.whatwg.org/#timestamp. Maybe we should keep it? I don't love "associated with" either.

Comment (hamishwillee) [reply files/en-us/web/api/serviceworkerregistration/shownotification/index.md:100]: FWIW I am OK with this and would have complained about "actual".
